docker/Dockerfile.dockerignore

```diff
@@ -0,0 +1 @@
+# Overrides top level .dockerignore to allow adding files to the image
```

Comment:
Can you explain this?  What top level file is being refereed too?    Can this file simply be call `.dockerignore` or does it need the longer name?

Reply:
It's the `.dockerignore` file in the root of the repository, which excludes all files since they aren't used in that build. This build does need files added.

The naming scheme is required to be like this for docker to pick it up as an override

Reply:
Can we instead just remove the top level one.. I don't know why that exists.

Reply:
Thanks @sbc100 - I've removed both. I suspect one motivation may have been the size of the submodules when initializing docker, but even on a slowish CI machine it seems to be just 4s so seems like an overoptimization

https://github.com/anuraaga/wasi-sdk/actions/runs/3728956980/jobs/6324422776#step:9:174

Reply:
Can you explain more?   Surly docker should not need to read the sub modules directory should it?  Doesn't it just include what we tell it to?

Reply:
Unfortunately no, docker begins by sending the entire context to the daemon - usually this is the current directory, `.` and is recursive. So to prevent transferring things during this step you need to use dockerignore, the values in ADD statements in the Dockerfile don't influence it.

Reply:
> Unfortunately no, docker begins by sending the entire context to the daemon - usually this is the current directory, `.` and is recursive. So to prevent transferring things during this step you need to use dockerignore, the values in ADD statements in the Dockerfile don't influence it.

That seems rather unfortunate!  So even if I only add one tiny file to the image the docker create command will be `O(size of everything my my CWD)`.   I did not know that.  

In that case maybe lets leave the top level `.dockerignore` but have it only ignore the large directories such as `dist`.

Reply:
Alternatively, can we run docker from the subdirectory and `ADD ../dist`?

Reply:
Nope, docker won't allow adding files that are't in the context.

> In that case maybe lets leave the top level .dockerignore but have it only ignore the large directories such as dist.

Since this image needs dist (to be able to package it), I think it would result in the two dockerignore files again, one ignoring dist at top level and the other not. Just to confirm that's ok?

Reply:
No, I think we should just have one top level dockerignore, and it should just specify the stuff we don't want (i.e. the submodules).

Reply:
Thanks @sbc100 - added a top level .dockerignore that excludes /src, the submodules folder.